### ISSUE TITLE
authentication: generate client keys on remote platforms

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#3443]
Follow on to #3359 

* Generate client keys on remote platforms rather than coppying them from the suite host.
* Once this is done private keys should not travel.

**Comment by sadielbartholomew** [comment]
> Once this is done private keys should not travel.

Agreed. This one is important.

**Comment by dwsutherland** [comment]
Please put this on hold until #3389 is merged, because the network code has been refactored (due to the new PUB/SUB pattern using the same security & logic).

**Comment by datamel** [comment]
Generate client keys on remote platforms #3608 merged.
